### ISSUE TITLE
Cannot query MySQL with hexadecimal

### ISSUE THREAD

**Issue by samijnih** [phalcon/cphalcon#10518]
Syntax error, unexpected token IDENTIFIER(x613A313A7B693A303B733A31333A226E656F20636C6173736963616C223B7D), near to ' ORDER BY name', when parsing: SELECT COUNT(42) AS rowcount FROM [Fymh\Api\Models\Band] WHERE activated = 1 AND styles = 0x613A313A7B693A303B733A31333A226E656F20636C6173736963616C223B7D ORDER BY name (168)

when 

    if ($styles && is_array($styles)) {
	    $bandsConditions[] = 'styles = ' . self::getStylesToHex($styles);
    }

    $infoConditions        = 'language = :language:';
    $infoBinds['language'] = $language;

    $bandsConditions = implode(' AND ', $bandsConditions);

    // Sets model's params
    $bandsParameters = [
	    'conditions' => !empty($bandsConditions) ? $bandsConditions : null,
	    'bind'       => !empty($bandsBinds) ? $bandsBinds : null,
	    'bindTypes'  => !empty($bandsBindsType) ? $bandsBindsType : null,
	    'order'      => $order,
    ];

**Comment by andresgutierrez** [comment]
You have to use bound parameters when comparing the styles column:

```php
if ($styles && is_array($styles)) {
    $bandsConditions[] = 'styles = :styles:';
    $bandBinds["styles"] = self::getStylesToHex($styles);
}
```

**Comment by samijnih** [comment]
Hi,
I already tested this but it produces me the same error

**Comment by andresgutierrez** [comment]
I don't think that use bound parameters produce the same error

**Comment by samijnih** [comment]
Using bound parameters doens't produce the same error but the query doesn't return me the result.

It's hexadecimal and boud parameters uses it as a string but it is not a string. I don't have to use bound parameters because my hexadecimal is 0x613A313A7B693A303B733A31333A226E656F20636C6173736963616C223B7D

I don't need to put quotes so that's my problem which gives you the problem

**Comment by andresgutierrez** [comment]
What column do you use to store a number like: 0x613A313A7B693A303B733A31333A226E656F20636C6173736963616C223B7D

**Comment by samijnih** [comment]
I serialize an array of gender of musique that I insert into a blob type

**Comment by dreamsxin** [comment]
If blod type, can bind type `PDO::PARAM_LOB`, but fllow you question can do this:
```php
if ($styles && is_array($styles)) {
    $bandsConditions[] = 'styles = HEX(UNHEX(:style:))';
    $bandBinds["styles"] = self::getStylesToHex($styles)
}
```

**Comment by samijnih** [comment]
ok for me

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
